src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

```diff
@@ -158,4 +159,35 @@ public String getTargetFileName(LinkedFile flEntry, BibEntry entry) {
     public int getUnsuccessfulRenames() {
         return unsuccessfulRenames;
     }
+
+    /**Check to see if a file already exists in the target directory.  Search is not case sensitive.
+    *
+    * @param flEntry
+    * @param entry
+    * @return First identified path that matches an existing file.  This name can be used in subsequent calls to override the existing file.
+    */
+    public Path fileAlreadyExists(LinkedFile flEntry, BibEntry entry) {
+        String targetFileName = getTargetFileName(flEntry, entry);
+        Path targetFilePath = flEntry.findIn(databaseContext,
+                fileDirectoryPreferences).get().getParent().resolve(targetFileName);
+        Path oldFilePath = flEntry.findIn(databaseContext, fileDirectoryPreferences).get();
+
+        //Check if file already exists in directory with different case.
+        //This is necessary because other entries may have such a file.
+        Path matchedByDiffCase = null;
+        try (Stream<Path> stream = Files.list(oldFilePath.getParent())) {
+            matchedByDiffCase = stream
+                    .filter(name -> name.toString().equalsIgnoreCase(targetFilePath.toString()))
+                    .findFirst()
+                    .get();
```

Comment:
findFirst also returns an optional, so you could just forward that one from here